### PR TITLE
Add Javadoc documentation to project

### PR DESCRIPTION
**Pull request by subhramit**
Closes https://github.com/LambdaTest/lambdatest-gradle-plugin/issues/8

Adds javadoc documentation to all classes and methods in the project (except for obvious methods such as setters in `LambdaTestTask`).

I have kept the documentation thorough, including how one class is dependent on another. Please read through and let me know if any additions/removals are necessary.

### CODE REVIEW

**Comment by subhramit** [comment]
I'm getting code format CI errors. I don't see why...

File: src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java [hunk 1]
```diff
@@ -12,11 +12,12 @@
...
- * This task manages the upload of applications and test suites, followed by test execution with specified configurations.
+ * Main task class for the LambdaTest Gradle plugin that handles test execution on the LambdaTest
+ * platform. This task manages the upload of applications and test suites, followed by test
+ * execution with specified configurations.
  *
- * <p>This task coordinates between {@link AppUploader}, {@link TestSuiteUploader}, and {@link TestExecutor}
- * to manage the complete test execution lifecycle.</p>
+ * <p>This task coordinates between {@link AppUploader}, {@link TestSuiteUploader}, and {@link
+ * TestExecutor} to manage the complete test execution lifecycle.
  */
```
Comment (subhramit) [inline src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java:21]: Spotless formatter broke this line... Not sure what to do

File: src/main/java/io/github/lambdatest/gradle/AppUploader.java [hunk 2]
```diff
@@ -6,10 +6,11 @@
 import org.apache.logging.log4j.Logger;
 
 /**
- * Handles the asynchronous upload of application files to the LambdaTest platform.
- * This class manages the upload process and returns the application ID for test execution.
+ * Handles the asynchronous upload of application files to the LambdaTest platform. This class
+ * manages the upload process and returns the application ID for test execution.
```
Comment (subhramit) [inline src/main/java/io/github/lambdatest/gradle/AppUploader.java:10]: Again, I think the formatter is broken. Let me know if I should undo the formatter actions (CI will complain, though).

Comment (Delta456) [reply src/main/java/io/github/lambdatest/gradle/AppUploader.java:10]: No, this is also fine.

**Comment by navin772** [comment]
@subhramit can you take a look at the output of `./gradlew javadoc`, it shows some errors which needs to be fixed.
One of the errors were - `error: unknown tag: implNote`, and other missing doc warnings.

Can you please fix those?

**Comment by subhramit** [comment]
> @subhramit can you take a look at the output of `./gradlew javadoc`, it shows some errors which needs to be fixed.
> One of the errors were - `error: unknown tag: implNote`, and other missing doc warnings.
> 
> Can you please fix those?

`@implNote`  (implementation note) is a common javadoc tag. I'll double check.

**Comment by subhramit** [comment]
Although `@implNote` expands properly in IntelliJ (if we are targeting developers to be able to navigate through the javadoc when contributing), I am changing @implnote to "Implementation note:" to fix the errors.
image

As a result of changes:
image

**Comment by subhramit** [comment]
@navin772 let me know if anything else is needed!

**Comment by subhramit** [comment]
> @subhramit is there any way we can have the `@implNote` so it renders as tag and we can ignore the `@implNote` specific errors. Or any other tag that can be used here?

Good point. Will do some reading on this and get back.

**Comment by subhramit** [comment]
@navin772 Explored a couple of options. This (https://github.com/LambdaTest/lambdatest-gradle-plugin/pull/18/commits/dcd1533609bb8d8544d17a79cda3831a29295e51) seems to be the best fit.

**Review by navin772 — APPROVED** [review]
LGTM!

**Pull request merged**